client/src/main/java/io/pravega/client/stream/Transaction.java

```diff
@@ -92,13 +92,16 @@
 
     /**
      * Drops the transaction, causing all events written to it to be deleted.
+     *
+     * @throws TxnFailedException Error aborting transaction: FAILURE
      */
-    void abort();
+    void abort() throws TxnFailedException;
 
     /**
      * Gets the status of the transaction.
      *
+     *  @throws TxnFailedException This exception could be thrown on checkStatus
      *  @return Current status of the transaction
      */
-    Status checkStatus();
+    Status checkStatus() throws TxnFailedException;
```

Comment:
This is weird from the user point of view.
TxnFailedException sounds like the transaction is broken or you broke it by calling this method.

Here I would expect some new TxNotValidException